app/src/main/java/com/example/swugger/EditEventDialogFragment.java

```diff
@@ -0,0 +1,313 @@
+import android.view.ViewGroup;
+import android.widget.DatePicker;
+import android.widget.EditText;
+import android.widget.ImageButton;
+import android.widget.ImageView;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+import android.widget.TimePicker;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Locale;
+
+public class EditEventDialogFragment extends DialogFragment implements DatePickerDialog.OnDateSetListener, TimePickerDialog.OnTimeSetListener, AddReminderDialogFragment.AddReminderDialogListener, RemindersAdapter.RemindersAdapterListener {
+
+    public static final long millisecondsPerDay = 86400000;
+    public static final long millisecondsPerHour = 3600000;
+    public static final long millisecondsPerMinute = 60000;
+    private EditEventDialogFragment thisDialog;
+    private EditEventsDialogListener mCallback;     // the callback fragment for THIS DoalogFragment
+    private Event mEvent;
+    private ImageButton mBackButton;
+    private ImageButton mSaveButton;
+    private RelativeLayout mDateLayout;
+    private RelativeLayout mTimeLayout;
+    private DatePicker mDatePicker;
+    private TimePicker mTimePicker;
+    private ImageView mDateIcon;
+    private ImageView mTimeIcon;
+    private ImageButton mAddRemindersBtn;
+    private RemindersAdapter mRemindersAdapter;
+    private ArrayList<Reminder> displayedRemindersList;
+    private ArrayList<Reminder> newRemindersList;
+    private ArrayList<ReminderWithId> remindersToDeleteList;
+    private TextView mDateText;
+    private TextView mTimeText;
+    private EditText mNameText;
+    private EditText mNotesText;
+
+    public EditEventDialogFragment() { }
+
+    @Override
+    public void onDateSet(DatePicker view, int year, int month, int dayOfMonth) {
+        mDatePicker = view;
+        // regardless of whether or not the user chose a different date, update the text
+        mDateText.setText(String.format(Locale.US, "%d/%d/%d", month+1, dayOfMonth, year));     // months+1 because months is zero-indexed
+        // change the text color to indicate whether or not there's a change
+        Resources res = getResources();
+        if (mEvent.getMonth() != month || mEvent.getDay() != dayOfMonth || mEvent.getYear() != year) {
+            mDateIcon.setColorFilter(res.getColor(R.color.colorPrimary));
+            mDateText.setTextColor(res.getColor(R.color.colorPrimary));
+        } else {
+            mDateIcon.setColorFilter(res.getColor(R.color.black));
+            mDateText.setTextColor(res.getColor(R.color.gray));
+        }
+    }
+
+    @Override
+    public void onTimeSet(TimePicker view, int hourOfDay, int minute) {
+        mTimePicker = view;
+        // regardless of whether or not the user chose a different time, update the text
+        mTimeText.setText(Event.convertHourAndMinuteToTimeStamp(hourOfDay, minute));
+        // change the text color to indicate whether or not there's a change
```

Comment:
Generally if your code needs basic comments like this you might want to consider renaming variables or making it a little bit more clear. 

Commenting is good, but also remember that if you need to make a comment about color changing then it's kinda poorly written

Reply:
the function of these lines is pretty clear; i just randomly comment often regardless of whether i read them later on because it helps act as a separator